pkg/controllers/proxy/network_services_controller.go

```diff
@@ -443,22 +442,27 @@ func (nsc *NetworkServicesController) doSync() error {
 	}
 
 	if nsc.MetricsEnabled {
-		nsc.publishMetrics(nsc.serviceMap)
+		err = nsc.publishMetrics(nsc.serviceMap)
+		glog.Errorf("Error publishing metrics: %s", err.Error())
```

Comment:
Not sure if the log is needed, it matches the style above but hopefully the caller should handle the error that gets returned.